web/2016-spring/git-editor-html/branch.md

```diff
@@ -0,0 +1,59 @@
+# Git Branch
+
+## What is branch?
+
+- Nearly every VCS has some form of branching support. 
+- Branching means you diverge from the main line of development and continue to do work without messing with that main line.
+
+see the following picture
+
+![image](./images/branch.png)
+
+## Working with branches
+
+### Create branch
+
+``git branch <branch name>``
+
+### Switch branch
+
+``git checkout <branch name>``
+
+It is worth mentioning that if you want to checkout to an inexistent branch, you will get error message. Like this one.
+
+```
+$ git checkout vain-branch
+error: pathspec 'vain-branch' did not match any file(s) known to git.
+```
+But if you add ``-b`` option, you will create a new branch and switch to the new branch
+```
+$ git checkout -b vain-branch
+Switched to a new branch 'vain-branch'
+```
+
```

Comment:
should mention how to checkout to remote branch and set upstream, git is having different behavior after 1.7 or 1.5. I don't remember the detail. cc @arrowrowe